content/documentation/sessionParameters/content.md

```diff
@@ -0,0 +1,84 @@
++++
+fragment = "content"
+weight = 100
+
+title = "How to inject custom parameters into a session"
+
+[sidebar]
+  sticky = true
++++
+
+This tutorial describes how custom parameters are injected into session pods as environment variables.
+This may be used to inject parameters into a session that are not equal for all sessions of an app definition.
+For instance, access tokens or dynamic parameters such as a Git repository to check out.
+
+## Adding environment variables via REST API
+
+The simplest way to start a Session with custom environment variables is using Theia Cloud's REST API.
+Theia Cloud offers npm package [@eclipse-theiacloud/common](https://www.npmjs.com/package/@eclipse-theiacloud/common) to access the REST API conveniently.
+It offers to provide environment variables directly or as lists of [config maps](https://kubernetes.io/docs/concepts/configuration/configmap/) or [secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to read in.
+
+The following code snippet show how to add environment variables while starting a session.
+Thereby, the whole `env` as well as its three properties `fromMap`, `fromConfigMaps` and `fromSecrets` are optional.
+The snippet assumes that the parameters `accessToken` and `user` are present as variables based on the authenticated user.
+You can have a look at the [Try Now Page](https://github.com/eclipsesource/theia-cloud/blob/main/node/try-now-page/src/App.tsx) to see an example of how these are derived when logging in via Keycloak.
+
+```typescript
+import { getTheiaCloudConfig, SessionStartRequest, TheiaCloud } from '@eclipse-theiacloud/common';
+
+// Get common parameters from Theia Cloud configuration
+const { appDefinition, appId, serviceUrl} = getTheiaCloudConfig();
+
+// Create request object with customized environment variables
+// user and accessToken are expected to be set based on the authenticated user
+const request: SessionStartRequest = {
+  serviceUrl,
+  accessToken,
+  appId,
+  user,
+  appDefinition,
+  env: {
+    fromMap: {
+      mykey: "myvalue"
+    },
+    fromConfigMaps: [ "session-config-map-1", "config-map-2"],
+    fromSecrets: [ "session-secret-1", "session-secret-2"]
+  }
+}
+
+// Send request
+TheiaCloud.Session.startSession(request);
+```
+
+## Adding environment variables via Session CR
+
+The environment variables can also be configured directly through the Session custom resource by adding one or multiple of the following properties:
+
+- [envVars](https://github.com/eclipsesource/theia-cloud-helm/blob/4cd9d98d30cebe8d31e7084369878c1c2d28776c/charts/theia-cloud-crds/templates/session-spec-resource.yaml#L46-L49): This property is a map that allows you to define environment variables directly. Each entry in the map consists of the environment variable name and its value. This method is suitable for standard configuration needs and facilitates individual configuration for each session.
```

Comment:
I don't like the links to specific commits, however I don't see a better option. 
Maybe we should keep track where such links are used, so we may check them regularly?

Reply:
Mhm, yes I agree that it is not optimal to use it like that. If we point to main, we would also need to keep track and check links regularly. Especially with references to CRDs where a new version will throw off the referenced lines completely.

Maybe we could keep a file like `repository-links` in the root of the repository that either keeps entries for every link and its file or just files that contain repository links. What do you think?

Reply:
Sounds good, I think keeping the files should be enough. 
I'll add it to the release template to check for outdated links after a release